src/CattyTests/Scene/PenFeatureTests.swift

```diff
@@ -0,0 +1,120 @@
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+import XCTest
+
+@testable import Pocket_Code
+
+final class PenFeatureTests: XCTestCase {
+
+    var scene: CBScene!
+    var spriteNode: CBSpriteNode!
+    var initialPosition: CGPoint!
+
+    override func setUp() {
+        spriteNode = CBSpriteNode(spriteObject: SpriteObject())
+        spriteNode.name = "testName"
+
+        scene = SceneBuilder(project: ProjectMock()).build()
+        scene.addChild(spriteNode)
+
+        initialPosition = CGPoint.zero
+    }
+
+    func testWhenSpritePositionChangedPenUp() {
+        spriteNode.penConfiguration.penDown = false
+        spriteNode.position = initialPosition
+
+        spriteNode.update(CACurrentMediaTime())
+        spriteNode.position = CGPoint(x: 1, y: 1)
+        spriteNode.update(CACurrentMediaTime())
+
+        XCTAssertEqual(scene.children.count, 1)
+    }
+
+    func testWhenSpritePositionChangedPenDown() {
+        spriteNode.penConfiguration.penDown = true
+        spriteNode.position = initialPosition
+
+        spriteNode.update(CACurrentMediaTime())
+        spriteNode.position = CGPoint(x: 1, y: 1)
+        spriteNode.update(CACurrentMediaTime())
+
+        XCTAssertEqual(scene.children.count, 2)
+
+        guard let line = scene.childNode(withName: SpriteKitDefines.penShapeNodeName) as? SKShapeNode else {
+            XCTFail("SKShapeNode for the drawn line not found with default name")
+            return
+        }
+
+        XCTAssertEqual(line.lineWidth, SpriteKitDefines.defaultPenSize, accuracy: 0.0001)
+        XCTAssertEqual(line.strokeColor, SpriteKitDefines.defaultPenColor)
+        XCTAssertEqual(line.path?.currentPoint, spriteNode.position)
+
+        spriteNode.position = CGPoint(x: 2, y: 2)
+        spriteNode.update(CACurrentMediaTime())
+
+        XCTAssertEqual(scene.children.count, 3)
+
+    }
+
+    func testWhenSpritePositionNotChangedPenUp() {
+        spriteNode.penConfiguration.penDown = false
+        spriteNode.position = initialPosition
+
+        spriteNode.update(CACurrentMediaTime())
+        spriteNode.update(CACurrentMediaTime())
+
+        XCTAssertEqual(scene.children.count, 1)
+    }
+
+    func testWhenSpritePositionNotChangedPenDown() {
+        spriteNode.penConfiguration.penDown = true
+        spriteNode.position = initialPosition
+
+        spriteNode.update(CACurrentMediaTime())
+        spriteNode.update(CACurrentMediaTime())
+
+        XCTAssertEqual(scene.children.count, 1)
```

Comment:
Good test! Just again please check that before executing the update method the number of children were equal to document that nothing has changed.